src/devices/machine/ldv4200hle.h

```diff
@@ -0,0 +1,215 @@
+// license:BSD-3-Clause
+// copyright-holders:Ryan Holtz
+/*************************************************************************
+
+    ldv1000hle.h
+
+    Pioneer LD-V4200 laserdisc player simulation.
+
+*************************************************************************/
+
+#ifndef MAME_MACHINE_LDV4200_H
+#define MAME_MACHINE_LDV4200_H
+
+#pragma once
+
+#include "laserdsc.h"
+#include "diserial.h"
+
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+// device type definition
+DECLARE_DEVICE_TYPE(PIONEER_LDV4200HLE, pioneer_ldv4200hle_device)
+
+
+
+//**************************************************************************
+//  GLOBAL HELPERS
+//**************************************************************************
+
+static constexpr uint16_t make_ldv4000_command(const char *str) { return ((uint8_t)str[0] << 8) | (uint8_t)str[1]; }
```

Comment:
Can you put this inside the class?  It should still work without being in the global namespace.

Reply:
I did, but it resulted in a "[make_ldv4000_command] called in a constant expression before its definition is complete" error for each entry in the enum list. I would have called it "make_command" were it in the class definition itself, but I added the "ldv4000" qualifier to at least reduce the chance of a global namespace collision. I've added a comment clarifying to this effect.